independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/codestarts/QuarkusCodestartTestBuilder.java

```diff
@@ -209,14 +209,12 @@ public QuarkusCodestartTestBuilder standaloneExtensionCatalog(String quarkusBomG
             Objects.requireNonNull(buildWithQuarkusCoreVersion, "quarkus version not found in extensions");
             String quarkusVersion = quarkusBomVersion != null ? quarkusBomVersion : buildWithQuarkusCoreVersion;
             enableRegistryClientTestConfig(quarkusBomGroupId != null ? quarkusBomGroupId : "io.quarkus", quarkusVersion);
-            final ExtensionCatalog extensionCatalog = QuarkusProjectHelper
-                    .getExtensionCatalog(quarkusVersion);
+            final ExtensionCatalog extensionCatalog = QuarkusProjectHelper.getExtensionCatalog(quarkusVersion);
             disableRegistryClientTestConfig();
-            if (!(extensionCatalog instanceof ExtensionCatalog.Mutable)) {
-                throw new IllegalStateException("Problem with the given ExtensionCatalog type");
-            }
-            extensions.addAll(extensionCatalog.getExtensions());
-            ((ExtensionCatalog.Mutable) extensionCatalog).setExtensions(extensions);
+            final ExtensionCatalog.Mutable mutableCatalog = extensionCatalog instanceof ExtensionCatalog.Mutable
```

Comment:
Can we make the `mutable()` return `this` in `ExtensionCatalog.Mutable` implementations? It feels wrong to check for an instanceof first

Reply:
mutable() acts as a copy constructor, "create a new thing based on this current thing". If mutable() returns `this`, that ability goes away.

Note that I'll be hitting all of this code again (in a second pass), trying to hide more things.. so it may be that this check goes away for other reasons...

Reply:
Perhaps there should also be `clone()`?

Reply:
And `mutable()` would return `this` for a mutable instance.

Reply:
Do you want to do that with this fix? Or defer to next round.

Reply:
I can but this there is also no rush if you like to think about it more. Let me know what you prefer.

Reply:
Let me do it in the next rev.. so I can make sure there aren't any knock-on effects.

Reply:
Agreed, thanks!